docs/wiki/EL/RLP.md

```diff
@@ -0,0 +1,198 @@
+  - After decoding the first item, the next byte in the sequence is another `0x83`.
+  - Following the same rule as before:
+    - `0x83 - 0x80 = 3`
+  - This indicates the next string also has a length of 3 bytes.
+  - The following three bytes are `0x64, 0x6f, 0x67`, corresponding to "dog".
+  - We have now decoded the second item: "dog".
+- The decoded output is `["cat", "dog"]`.
+
+## How RLP Algorithm works
+
+**RLP encoding algorithm**
+
+Here is a flow chart describing how RLP encoding algorithm works.
+
+_Note that in some RLP tools, some clients may add additional conditional cases to the flow. These additional cases are not part of the standard specification but they are useful for the clients for the correct data serialization, e.g. geth client node communicating with a Nethermind client node._
+
+```mermaid
+flowchart TD
+    A[Start: Input] --> B{Is Input null, empty string, or false?}
+    B -->|Yes| C[Encode as Single Byte 0x80]
+    B -->|No| D{Is it a Single Byte and < 0x80?}
+    D -->|Yes| E[Directly Encode the Byte]
+    D -->|No| F{Is it a String or List?}
+    F -->|String| G{String Length <= 55}
+    G -->|Yes| H[Encode with Length + 0x80]
+    G -->|No| I[Encode Length with 0xB7 + Length Bytes]
+    F -->|List| J{Total Encoded Items Length <= 55}
+    J -->|Yes| K[Encode with Total Length + 0xC0]
+    J -->|No| L[Encode Total Length with 0xF7 + Length Bytes]
+    H --> M[Output Encoded Data]
+    I --> M
+    K --> M
+    L --> M
+    E --> M
+    C --> M
+```
+
+_Figure: RLP Encoding Flow_
+
+**RLP decoding algorithm**
+
+Here is a flow chart describing how RLP decoding algorithm works.
+
+```mermaid
+flowchart TD
+    A[Start: Encoded Input] --> B{Check First Byte}
+    B -->|0x00 to 0x7F| C[Direct Byte Output]
+    B -->|0x80| D[Decode as Empty String/Null/False]
+    B -->|0x81 to 0xB7| E[Short String]
+    E --> F[Subtract 0x80 from First Byte to get Length]
+    F --> G[Output Next 'Length' Bytes as String]
+    B -->|0xB8 to 0xBF| H[Long String]
+    H --> I[Subtract 0xB7 from First Byte to get Length Bytes Count]
+    I --> J[Read Next 'Length Bytes Count' to Determine String Length]
+    J --> K[Output Next 'String Length' Bytes as String]
+    B -->|0xC0 to 0xF7| L[Short List]
+    L --> M[Subtract 0xC0 from First Byte to get Total Encoded Length]
+    M --> N[Decode Each Element in the Next 'Total Encoded Length' Bytes Recursively]
+    B -->|0xF8 to 0xFF| O[Long List]
+    O --> P[Subtract 0xF7 from First Byte to get Length Bytes Count]
+    P --> Q[Read Next 'Length Bytes Count' to Determine List Total Length]
+    Q --> R[Decode Each Element in the Next 'List Total Length' Bytes Recursively]
+    C --> S[Output Decoded Data]
+    D --> S
+    G --> S
+    K --> S
+    N --> S
+    R --> S
+```
+
+_Figure: RLP Decoding Flow_
+
+
+## RLP Tools
+
+There are many libraries available for RLP implementations in Ethereum. Here are few tools:
+- [Geth RLP](https://github.com/ethereum/go-ethereum/tree/master/rlp)
```

Comment:
Apart from the decoding library in geth, here is an actual tool for decoding https://github.com/ethereum/go-ethereum/tree/master/cmd/rlpdump